### PR TITLE
Updates for Singularity documentation - addressing bind mount potential issues.

### PR DESCRIPTION
**Pull request by jkwmoore**
Added warning boxes where I think is appropriate - we may wish to tinker with the  wording for the bind mount warning itself however.

### CODE REVIEW

File: bessemer/software/apps/singularity.rst [hunk 1]
```diff
@@ -156,6 +156,18 @@ When running Singularity images on Bessemer,
 the paths ``/fastdata``, ``/home``, ``/scratch``, ``/shared`` are
 automatically mounted to your Bessemer directories.
 
+.. warning::
+
+  The automatic bind mounting of the ``/home`` path can result in shell environment corruption and unintended sharing of executables
```
Comment (willfurnass) [inline bessemer/software/apps/singularity.rst:161]:
```suggestion
  The automatic bind mounting of your HPC home directory into Singularity containers can result in the unexpected sharing of things like executables 
```

File: bessemer/software/apps/singularity.rst [hunk 2]
```diff
@@ -156,6 +156,18 @@ When running Singularity images on Bessemer,
 the paths ``/fastdata``, ``/home``, ``/scratch``, ``/shared`` are
 automatically mounted to your Bessemer directories.
 
+.. warning::
+
+  The automatic bind mounting of the ``/home`` path can result in shell environment corruption and unintended sharing of executables 
+  and libraries between the host and Singularity container.
+
+  Unintended behaviour / corrupted Singularity environments may occur due to the presence of:
```
Comment (willfurnass) [inline bessemer/software/apps/singularity.rst:164]:
```suggestion
  Unintended behaviour may occur with Singularity on the HPC system due to the presence of:
```

File: bessemer/software/apps/singularity.rst [hunk 3]
```diff
@@ -156,6 +156,18 @@ When running Singularity images on Bessemer,
...
+
+  * Shell initialisation files e.g. ``~/.bashrc`` or ``~/.profile``
+  * R profile files or libraries e.g. ``~/.Rprofile``
```
Comment (willfurnass) [inline bessemer/software/apps/singularity.rst:167]:
```suggestion
  * R profile files (e.g. ``~/.Rprofile``) and/or libraries (e.g ``~/R/x86_64-pc-linux-gnu-library/4.1``)
```

**Review by willfurnass — CHANGES_REQUESTED** [review]
Suggested a few small changes to wording.  Not keen on the term 'environment corruption' but that might be a personal thing!

File: sharc/software/apps/singularity.rst [hunk 4]
```diff
@@ -165,6 +165,18 @@ automatically *bind-mounted* (exposed) from the *host* operating system into you
 i.e. the cluster's ordinary filestores will be automatically visible within a container started on the cluster
 without that directory being explicitly created when the corresponding Singularity image was built.
 
+.. warning::
+
+  The automatic bind mounting of the ``/home`` path can result in shell environment corruption and unintended sharing of executables
```
Comment (willfurnass) [inline sharc/software/apps/singularity.rst:170]: Suggest making similar edits to this section for consistency

Comment (jkwmoore) [reply sharc/software/apps/singularity.rst]: Just sorted.

**Comment by jkwmoore** [comment]
Unlinked the issue for now so we can keep it open for further work around these pages.

File: sharc/software/apps/singularity.rst [hunk 5]
```diff
@@ -165,6 +165,18 @@ automatically *bind-mounted* (exposed) from the *host* operating system into you
...
 
+.. warning::
+
+  The automatic bind mounting of your HPC home directory into Singularity containers can result in the unexpected sharing of things like executables 
+  and libraries between the host and Singularity container.
+
+  Unintended behaviour / corrupted Singularity environments may occur due to the presence of:
+
+  * Shell initialisation files e.g. ``~/.bashrc`` or ``~/.profile``
+  * R profile files or libraries e.g. ``~/.Rprofile``
```
Comment (willfurnass) [inline sharc/software/apps/singularity.rst:176]:
```suggestion
  * R profile files (e.g. ``~/.Rprofile``) and/or libraries (e.g ``~/R/x86_64-pc-linux-gnu-library/4.1``)
```

File: sharc/software/apps/singularity.rst [hunk 6]
```diff
@@ -165,6 +165,18 @@ automatically *bind-mounted* (exposed) from the *host* operating system into you
 i.e. the cluster's ordinary filestores will be automatically visible within a container started on the cluster
 without that directory being explicitly created when the corresponding Singularity image was built.
 
+.. warning::
+
+  The automatic bind mounting of your HPC home directory into Singularity containers can result in the unexpected sharing of things like executables 
+  and libraries between the host and Singularity container.
+
+  Unintended behaviour / corrupted Singularity environments may occur due to the presence of:
```
Comment (willfurnass) [inline sharc/software/apps/singularity.rst:173]:
```suggestion
  Unintended behaviour may occur with Singularity on the HPC system due to the presence of:
```

**Review by willfurnass — CHANGES_REQUESTED** [review]

**Comment by willfurnass** [comment]
Suggested a couple more changes.  Happy to merge after that.

File: sharc/software/apps/singularity.rst (hunk 5 above)
Comment (jkwmoore) [reply sharc/software/apps/singularity.rst:176]: Whoops - we did edit more than just the starter paragraph.

**Comment by jkwmoore** [comment]
> Suggested a couple more changes. Happy to merge after that.

Yep, my bad - I only matched the first bit >_<

**Review by willfurnass — APPROVED** [review]

**Pull request merged**